zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/encoding/AuthInfo.java

```diff
@@ -53,15 +56,36 @@ public AuthInfo setAuthData(final String authData) {
     return this;
   }
 
+  public Map<String, Object> getClaims() {
+    return MsgPackConverter.convertToMap(claimsProp.getValue());
+  }
+
+  public AuthInfo setClaims(final DirectBuffer authInfo) {
+    claimsProp.setValue(authInfo);
+    return this;
+  }
+
+  public AuthInfo setClaims(final Map<String, Object> authInfo) {
+    claimsProp.setValue(new UnsafeBuffer(MsgPackConverter.convertToMsgPack(authInfo)));
+    return this;
+  }
+
+  @JsonIgnore
+  public DirectBuffer getClaimsBuffer() {
+    return claimsProp.getValue();
+  }
+
   public void wrap(final AuthInfo authInfo) {
     formatProp.setValue(authInfo.getFormat());
     authDataProp.setValue(authInfo.getAuthData());
+    claimsProp.setValue(authInfo.getClaimsBuffer());
```

Comment:
:thought_balloon: Please double-check whether this is truly a `wrap`, i.e. we want to wrap over the same mutable byte buffers, or actually meant as a copy. `getAuthData` always copies, so it's possible that we need to do the same for the claims document.

Reply:
As far as I can see, the `wrap` method is used as a copy method.